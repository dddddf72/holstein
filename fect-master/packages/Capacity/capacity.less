.fect-capacity {
  width: 50px;
  height: 10px;
  overflow: hidden;
  background-color: var(--accents-2);
  border-radius: 5px;
  span {
    height: 100%;
    display: block;
    margin: 0;
    // width: 10%;
    padding: 0;
    // width: 100%;
    // background-color: ;
  }
}
