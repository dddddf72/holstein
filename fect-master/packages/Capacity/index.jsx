import { computed, toRefs } from 'vue'
import { createNameSpace } from '../utils'
const [createComponent] = createNameSpace('Capacity')
import './capacity.less'

export default createComponent({
  props: {
    value: {
      type: [Number, String],
      default: 0,
    },
    limit: {
      type: [Number, String],
      default: 0,
    },
    color: String,
  },
  setup(props, { attrs }) {
    const { value, limit, color } = toRefs(props)
    const calcColor = computed(() => {
      if (color?.value) {
        return color.value
      }
      const val = value.value
      if (val < 33) return 'var(--hightlight-cyan)'
      if (val < 66) return 'var(--warning-light)'
      return 'var(--error-dark)'
    })

    return () => (
      <>
        <div
          {...attrs}
          title={`${value.value}%`}
          className={`fect-capacity ${attrs.class ? attrs.class : ''}`}
        >
          <span
            style={{
              backgroundColor: calcColor.value,
              width: `${value.value}%`,
            }}
          ></span>
        </div>
      </>
    )
  },
})
