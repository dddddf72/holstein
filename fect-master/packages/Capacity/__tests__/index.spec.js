import { mount } from '@vue/test-utils'
import Capacity from '../index'

describe('Capacity', () => {
  it('should be render as element', () => {
    const capacity = mount(Capacity)
    expect(() => capacity.unmount()).not.toThrow()
  })

  it('should render title attrs', () => {
    const wrapper = mount({
      render(h) {
        return <Capacity value="40" />
      },
    })
    const el = wrapper.find('.fect-capacity')
    expect(el.attributes('title')).toBe('40%')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should support custom color', () => {
    const wrapper = mount({
      render(h) {
        return <Capacity value={'30'} color="pink" />
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
