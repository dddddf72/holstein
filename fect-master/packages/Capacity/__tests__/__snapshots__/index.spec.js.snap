// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Capacity should render title attrs 1`] = `<div title="40%" class="fect-capacity "><span style="width: 40%;"></span></div>`;

exports[`Capacity should support custom color 1`] = `<div title="30%" class="fect-capacity "><span style="background-color: pink; width: 30%;"></span></div>`;
